framework/src/st_things/things_stack/src/common/framework/things_data_manager.c

```diff
@@ -0,0 +1,2345 @@
+
+static struct hashmap_s *g_resource_type_hmap = NULL;	// map for resource types
+static struct hashmap_s *g_device_hmap = NULL;
+
+static bool is_there_cloud_data = false;
+static const char *origin_cloud_json_str = "{\n\
+		\"cloud\":    {\n\
+			\"address\":    \"52.40.216.160:5683\"\n\
+		},\n\
+		\"wifi\": {\n\
+			\"ssid\": \"unknown\"\n\
+		}\n\
+}";
+
+static wifi_manager_softap_config_s ap_config;
+static easysetup_connectivity_type_e es_conn_type = es_conn_type_none;
+static wifi_manager_ap_config_s g_homeap_info;
+
+typedef int8_t INT8;
+
+#define STD_CI_CLOUD_ADDRESS "52.40.216.160:5683"
+#define MAX_CI_CLOUD_ADDRESS 256
+static es_cloud_signup_s *gpst_cloud_data = NULL;
+static char g_ci_cloud_address[MAX_CI_CLOUD_ADDRESS] = { 0 };
+
+struct st_attribute_s const gstAttr_x_com_samsung_provisioninginfo[] = {
+	{
+		.key = "x.com.samsung.provisioning.targets",
+		.type = 9,
+		.mandatory = false,
+		.rw = 1
+	},
+	{
+		.key = "x.com.samsung.provisioning.owned",
+		.type = 0,
+		.mandatory = false,
+		.rw = 1
+	},
+	{
+		.key = "x.com.samsung.provisioning.easysetupdi",
+		.type = 3,
+		.mandatory = false,
+		.rw = 1
+	},
+	{
+		.key = "x.com.samsung.provisioning.reset",
+		.type = 0,
+		.mandatory = false,
+		.rw = 3
+	},
+	{
+		.key = "x.com.samsung.provisioning.stopAP",
+		.type = 0,
+		.mandatory = false,
+		.rw = 3
+	}
+};
+
+const struct st_attribute_s const gstAttr_x_com_samsung_accesspointlist[] = {
+	{
+		.key = "x.com.samsung.accesspoint.items",
+		.type = 9,
+		.mandatory = true,
+		.rw = 1
+	}
+};
+
+static const struct st_resource_type_s const gst_resource_types[] = {
+	{
+		.rt = "x.com.samsung.provisioninginfo",
+		.prop_cnt = 5,
+		.prop[0] = &gstAttr_x_com_samsung_provisioninginfo[0],
+		.prop[1] = &gstAttr_x_com_samsung_provisioninginfo[1],
+		.prop[2] = &gstAttr_x_com_samsung_provisioninginfo[2],
+		.prop[3] = &gstAttr_x_com_samsung_provisioninginfo[3],
+		.prop[4] = &gstAttr_x_com_samsung_provisioninginfo[4]
+	},
+	{
+		.rt = "x.com.samsung.accesspointlist",
+		.prop_cnt = 1,
+		.prop[0] = &gstAttr_x_com_samsung_accesspointlist[0]
+	}
+};
+
+static const struct st_resource_s const gstResources[] = {
+	{
+		.uri = "/sec/provisioninginfo",
+		.interface_types = {"oic.if.a"},
+		.resource_types = {"x.com.samsung.provisioninginfo"},
+		.if_cnt = 1,
+		.rt_cnt = 1,
+		.policy = 3
+	},
+	{
+		.uri = "/sec/accesspointlist",
+		.interface_types = {"oic.if.s"},
+		.resource_types = {"x.com.samsung.accesspointlist"},
+		.if_cnt = 1,
+		.rt_cnt = 1,
+		.policy = 3
+	}
+};
+
+bool util_str_copy(char **dest, const char *src)
+{
+	if (NULL == src || NULL == dest) {
+		return false;
+	}
+
+	int srcLen = strlen(src);
+
+	(*dest) = (char *)things_malloc(srcLen + 1);
+	if (NULL == *dest) {
+		THINGS_LOG_D_ERROR(THINGS_ERROR, TAG, "Malloc failed");
+		return false;
+	}
+
+	memcpy((*dest), src, srcLen + 1);
+	return true;
+}
+
+bool update_and_save_device_info(st_device_s **dev_list, int dev_cnt)
+{
+	assert(1);
```

Comment:
It seems that assert(1) is useless.
Also this function always returns false in "dm_update_device_iInfo_list()" at line 2010 and then return again at line 1791.
But the return value is not used anymore. We may remove line 1790~1792 and this func. + "dm_update_device_iInfo_list()".
Please see also the local variable "write_file_for_device_id" seems to be useless.

Reply:
reflected your requirements.